gems/aws-eventstream/lib/aws-eventstream/bytes_buffer.rb

```diff
@@ -7,7 +7,7 @@ class BytesBuffer
       # This Util class is for Decoder/Encoder usage only
       # Not for public common bytes buffer usage
       def initialize(data)
-        @data = data
+        @data = data.dup
```

Comment:
I'm assuming this was necessary because we are manipulating `@data` in this file? What would need to be changed so that `dup` isn't necessary? Ditto on the other `dup` usages.

Reply:
I will try to figure that out.  
To be honest, the code around buffers is quite messy. I don't understand everything yet.